packages/block-library/src/group/theme.scss

```diff
@@ -6,4 +6,181 @@
 		margin-top: 0;
 		margin-bottom: 0;
 	}
+
```

Comment:
This is going to be really difficult for themes to override if they need to: there are a ton of different breakpoint/margin/padding combinations, and I worry that the theme would essentially need to rewrite all of them. This _should_ work out of the box in many themes (it seems to be okay in Twenty Nineteen based on my testing), but will definitely break some too. For instance, allowing users to adjust the margins in the Gutenberg starter theme causes it to completely fall apart, since it relies on all blocks having `auto` left and right margins. For that reason, I wonder if this is this "Allow users to control margin and padding for individual blocks" is something that themes opt-in to? Similar to how they opt into wide alignments currently. 

On a related note, most themes won't actually use the exact same breakpoints we use here. In #13363, we talked about potentially allowing themes to specify breakpoints for each of these devices. I'm not sure how/if that would best be implemented though. 🤔

Reply:
> This is going to be really difficult for themes to override if they need to

I've got to be honest, I've kind of hacked this in, in the hope that someone more famliar with Themes can advise on how best to implement to keep specificity down and allow Themes to extend. I'd greatly value any commits you'd like to push in this direction.

> On a related note, most themes won't actually use the exact same breakpoints we use here

Oh god, I didn't think of that. Of course, the Block Editor uses standard breakpoints, but Themes will choose their own.

I agree that we need a way to allow users to opt-in and to specify breakpoints. This seems like a wider project. 

What route do we have to allow an initial MVP implementation of Block spacing to be shipped? Or do we need to get the breakpoint registration stuff done first?

Thanks!

Reply:
> I've got to be honest, I've kind of hacked this in, in the hope that someone more famliar with Themes can advise on how best to implement to keep specificity down and allow Themes to extend. I'd greatly value any commits you'd like to push in this direction.

I'll have to think about this a little bit. I wonder if @allancole, @laurelfulford, or @davidakennedy has any thoughts too. In general, I think we'll have much more leeway if we make this an opt-in for themes. 

> What route do we have to allow an initial MVP implementation of Block spacing to be shipped? Or do we need to get the breakpoint registration stuff done first?

I wonder if shipping the automatically-responsive spacing makes the most sense as an MVP. Then we can tackle the breakpoint-specific settings alongside theme-supplied breakpoints in a separate, wider issue.

Reply:
> I'll have to think about this a little bit

Thanks 👍 

> I wonder if shipping the automatically-responsive spacing makes the most sense as an MVP. Then we can tackle the breakpoint-specific settings alongside theme-supplied breakpoints in a separate, wider issue.

Agreed and I think this is what I will do. So MVP will ship with basic controls (no responsive). Then I'll work on branches for 

* a generic <ResponsiveControls /> component (as seen here but not extracted)
* the `<DimensionControl />` component (as seen here)

Then when we've had a chance to think about the impact of responsive settings on Themes we can look to bring it all together.

Reply:
Why is this in the `theme.scss` file if it is an editor-only style?